README.md

```diff
@@ -1375,6 +1375,15 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
+* <a id='no_direct_standard_out'></a>(<a href='#no_direct_standard_out'>link</a>) **Prefer dedicated logging systems like [`os_log`](https://developer.apple.com/documentation/os/logging) or [`swift-log`](https://github.com/apple/swift-log) over writing directly to standard out using `print(…)`, `logDebug(…)`, or `dump(…)`.**
```

Comment:
For a Swift CLI tool, `print(...)` is appropriate (as I understand). Is the intention here to allow that use case, since writing to STDOUT is not "logging" in a CLI tool?

Reply:
Good point—even in a Swift CLI I'd imagine it would be a best practice to support dynamic levels of output, e.g. a `--quiet` mode or a `--verbose` mode that includes extra levels of output. Just calling `print` directly feels like it would still be a code small. What do you think?

Reply:
I agree that `--quiet` and `--verbose` modes are great additions to a robust CLI tool. Those modes would impact how much output you ultimately sent to STDOUT. When you have decided what output to send to STDOUT though, wouldn't you need to invoke `print(...)` to actually write the output to the STDOUT stream?  `print(...)` also allows you to send output to STDERR by specifying a text output stream.

<img width="1190" alt="Screen Shot 2021-04-22 at 5 58 12 PM" src="https://user-images.githubusercontent.com/1791049/115802753-5024ac00-a394-11eb-92a9-f591588132c6.png">

Reply:
> When you have decided what output to send to STDOUT though, wouldn't you need to invoke print(...) to actually write the output to the STDOUT stream

I think you'd probably `write` with a formatted log string on a standard out `TextOutputStream` directly, which is what swift-log does:

https://github.com/apple/swift-log/blob/3577a992d373d0e7f75d1fdc6e6e4570b60a1e1f/Sources/Logging/Logging.swift#L1124

However if you do end up wanting to call `print` directly in your logging system, it should be possible to silence the warning. Perhaps we could update the rule "details" to specify that only the logging system should write directly to standard out?

Reply:
Interesting, I hadn't considered `write(...)`. This rule makes more sense to me now that I see `print(...)` is under "Debugging and Reflection".
![image](https://user-images.githubusercontent.com/1791049/115813796-b0254d80-a3a8-11eb-8934-977f398ee8aa.jpeg)

Reply:
I _think_ that I am OK with this rule as-is now that I have learned more about the available API. As the rule is written I think it would be reasonable for a CLI tool to `write(...)` core output to STDOUT outside of a logging system, since core output wouldn't be considered a log in my mind. I wouldn't consider the text outputted by `echo` to be a log per se, and it seems like it would be OK for `echo` to directly use `write(...)` without a logging system, in my interpretation of this rule. Does this align with your intention?

Reply:
If so, maybe the rule slug could be `no-direct-log-output`?